Sources/MessagingPush/MessagingPushImplementation.swift

```diff
@@ -43,6 +47,14 @@ class MessagingPushImplementation: MessagingPushInstance {
         eventBusHandler.postEvent(TrackMetricEvent(deliveryID: deliveryID, event: event.rawValue, deviceToken: deviceToken))
     }
 
+    func trackMetricFromNSE(
```

Comment:
```suggestion
    func trackMetricOrFail(
```

From a code maintenance standpoint, I suggest we name functions by what they do (attempt or track metric, or fail) instead of where they are used (NSE). 

I think it could be ever better to isolate the NSE logic from the host app logic since the use case is different. 

Example: 
```swift
// Used by host app
class MessagingPushImplementation {
  trackMetric()
    eventBus...
}

// Used by NSE
class MessagingPushNSEImplementation: MessagingPushImplementation {
  override trackMetric()
    httpClient...
}
```

Reply:
Although I don't like having 2 `trackMetrics` functions with different bodies, we know we have more refactors coming for the NSE target. Therefore, I am OK with holding off on change to then.